src/Lecture3.hs

```diff
@@ -178,7 +199,13 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
-appendDiff3 = error "TODO"
+appendDiff3 :: (Semigroup a, Eq a) => a -> a -> a -> a
+appendDiff3 a b c
+  | a == b && a == c = a
+  | a == b && b /= c = a <> c
+  | a == c && b /= c = a <> b
```

Comment:
You can notice that you don't need extra checks here. If you rewrite your first check as:

```haskell
a == b && b == c
```

You can see that `b /= c` is automatically a possibility if `a == b` but not `a == b && b == c`
```suggestion
  | a == b           = a <> c
  | a == c           = a <> b
```